Tests/AK/TestFuzzyMatch.cpp

```diff
@@ -60,11 +58,16 @@ TEST_CASE(is_separator_bonus_applied_correctly)
     // Separator bonus is +30 if the character preceding the matching character is a space or an underscore.
     EXPECT_EQ(fuzzy_match("b"sv, "a b"sv).score, 118);
     EXPECT_EQ(fuzzy_match("bc"sv, "a b c"sv).score, 147);
-    EXPECT_EQ(fuzzy_match("ab cd"sv, "ab cd"sv).score, 205);
-    EXPECT_EQ(fuzzy_match("ab_cd"sv, "ab_cd"sv).score, 205);
     EXPECT_EQ(fuzzy_match("abcd"sv, "a b c d"sv).score, 202);
     EXPECT_EQ(fuzzy_match("abcd"sv, "a_b_c_d"sv).score, 202);
     EXPECT_EQ(fuzzy_match("b c"sv, "ab cd"sv).score, 153);
     EXPECT_EQ(fuzzy_match("b_c"sv, "ab_cd"sv).score, 153);
     EXPECT_EQ(fuzzy_match("bc"sv, "ab cd"sv).score, 122);
 }
+
+TEST_CASE(equality)
+{
+
+    EXPECT_EQ(fuzzy_match("abc"sv, "abc"sv).score, NumericLimits<int>::max());
```

Comment:
Is it important that it's equal to `NumericLimits<int>::max()`? Or is just the line after it sufficient?

Reply:
IMO result for equality should not be less than `NumericLimits<decltype(fuzzy_match(sv, sv))>::max()`. If it is less, there might be a non-equal string with a higher score. So this is the only case where the value actually matter.

Reply:
That's in theory possible for `NumericLimits<decltype(fuzzy_match(sv, sv))>::max()`, and in practice won't happen for many other values other than exactly `max()`, no?

Reply:
(To be clear, fine to keep as-is if you like it as-is.)

Reply:
True, the max value will still be an "issue", so it doesn't even add much theoretical value. I can't modify it before tonight though.